akka-persistence/src/main/scala/akka/persistence/journal/JournalProxy.scala

```diff
@@ -0,0 +1,181 @@
+      throw new IllegalArgumentException("Unknown plugin type: " + other)
+  }
+
+  private val timeout: FiniteDuration = config.getDuration("init-timeout", MILLISECONDS).millis
+  private val targetPluginId: String = {
+    val key = s"target-${pluginType.qualifier}-plugin"
+    config.getString(key).requiring(_ != "", s"$pluginId.$key must be defined")
+  }
+  private val startTarget: Boolean = config.getBoolean(s"start-target-${pluginType.qualifier}")
+
+  override def preStart(): Unit = {
+    if (startTarget) {
+      val target = pluginType match {
+        case Journal ⇒
+          log.info("Starting target journal [{}]", targetPluginId)
+          Persistence(context.system).journalFor(targetPluginId)
+        case SnapshotStore ⇒
+          log.info("Starting target snapshot-store [{}]", targetPluginId)
+          Persistence(context.system).snapshotStoreFor(targetPluginId)
+      }
+      context.become(active(target, targetAtThisNode = true))
+    } else {
+      context.system.scheduler.scheduleOnce(timeout, self, InitTimeout)(context.dispatcher)
+    }
+  }
+
+  private val selfAddress: Address =
+    context.system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+
+  private def timeoutException() = new TimeoutException(s"Target ${pluginType.qualifier} not initialized. " +
+    "Use `JournalProxy.setTargetLocation`")
+
+  def receive = init
+
+  def init: Receive = {
+    case TargetLocation(address) ⇒
+      context.setReceiveTimeout(1.second) // for retries
+      context.become(identifying(address))
+    case InitTimeout ⇒
+      log.info("Initialization timeout, Use `JournalProxy.setTargetLocation`")
+      context.become(initTimedOut)
+      unstashAll() // will trigger appropriate failures
+    case msg ⇒
+      stash()
+  }
+
+  def becomeIdentifying(address: Address): Unit = {
+    sendIdentify(address)
+    context.setReceiveTimeout(1.second) // for retries
+    context.become(identifying(address))
+  }
+
+  def sendIdentify(address: Address): Unit = {
+    val sel = context.actorSelection(RootActorPath(address) / "system" / targetPluginId)
+    log.info("Trying to identify target {} at {}", pluginType.qualifier, sel)
+    sel ! Identify(targetPluginId)
+  }
+
+  def identifying(address: Address): Receive = ({
+    case ActorIdentity(`targetPluginId`, Some(target)) ⇒
+      log.info("Found target {} at [{}]", pluginType.qualifier, address)
+      context.setReceiveTimeout(Duration.Undefined)
+      unstashAll()
+      context.become(active(target, address == selfAddress))
+    case _: ActorIdentity ⇒ // will retry after ReceiveTimeout
+    case ReceiveTimeout ⇒
+      sendIdentify(address)
+  }: Receive).orElse(init)
+
+  def active(targetJournal: ActorRef, targetAtThisNode: Boolean): Receive = {
+    case TargetLocation(address) ⇒
+      if (targetAtThisNode && address != selfAddress)
+        becomeIdentifying(address)
+    case InitTimeout ⇒
+    case msg ⇒
+      targetJournal.forward(msg)
```

Comment:
Good point. I'll see if I can use ask instead. (we can't resend, since there is no deduplication)

Long term, we have another ticket that would solve this at the right place: https://github.com/akka/akka/issues/15303

It should probably also watch the target journal.